substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jfr/Target_jdk_jfr_internal_SecuritySupport.java

```diff
@@ -30,24 +30,25 @@
 import com.oracle.svm.core.annotate.RecomputeFieldValue;
 import com.oracle.svm.core.annotate.Substitute;
 import com.oracle.svm.core.annotate.TargetClass;
+import com.oracle.svm.core.annotate.TargetElement;
 import com.oracle.svm.core.util.VMError;
 
-import jdk.jfr.internal.SecuritySupport.SafePath;
+import jdk.jfr.internal.SecuritySupport;
 
 @TargetClass(value = jdk.jfr.internal.SecuritySupport.class, onlyWith = HasJfrSupport.class)
 public final class Target_jdk_jfr_internal_SecuritySupport {
     // Checkstyle: stop
-    @Alias @RecomputeFieldValue(kind = RecomputeFieldValue.Kind.Reset) //
-    static SafePath JFC_DIRECTORY;
-    @Alias @RecomputeFieldValue(kind = RecomputeFieldValue.Kind.Reset) //
-    static SafePath JAVA_IO_TMPDIR;
+    @Alias @TargetElement(onlyWith = SecuritySupportSafePathPresent.class) @RecomputeFieldValue(kind = RecomputeFieldValue.Kind.Reset) //
+    static Target_jdk_jfr_internal_SecuritySupport_SafePath JFC_DIRECTORY;
+    @Alias @TargetElement(onlyWith = SecuritySupportSafePathPresent.class) @RecomputeFieldValue(kind = RecomputeFieldValue.Kind.Reset) //
+    static Target_jdk_jfr_internal_SecuritySupport_SafePath JAVA_IO_TMPDIR;
     // Checkstyle: resume
 
-    @Substitute
-    public static List<SafePath> getPredefinedJFCFiles() {
+    @Substitute @TargetElement(onlyWith = SecuritySupportSafePathPresent.class)
+    public static List<Target_jdk_jfr_internal_SecuritySupport_SafePath> getPredefinedJFCFiles() {
         throw VMError.shouldNotReachHere("Paths from the image build must not be embedded into the Native Image.");
     }
 
-    @Alias
-    static native SafePath getPathInProperty(String prop, String subPath);
+    @Alias @TargetElement(onlyWith = SecuritySupportSafePathPresent.class)
```

Comment:
I think all these `@TargetElement(onlyWith = SecuritySupportSafePathPresent.class)` can be applied to the  `@TargetClass` instead to make things a bit simpler